readthedocs/search/documents.py

```diff
@@ -88,17 +87,14 @@ class PageDocument(RTDDocTypeMixin, DocType):
             'role_name': fields.KeywordField(),
 
             # For linking to the URL
-            'doc_name': fields.KeywordField(),
             'anchor': fields.KeywordField(),
 
             # For showing in the search result
-            'type_display': fields.TextField(),
-            'doc_display': fields.TextField(),
```

Comment:
Why are we removing this?

Reply:
We don't need `doc_display` -- because the main title of each result contains the `doc_display`.
But I have undoed the change for `type_display` -- we can make use of this.

Reply:
:confused: 
I can't seem to find a good way to display every information.
`type_display` looks unncessary when we are displaying `role_name`.

Reply:
Well, sure. They are showing the same information. I believe the `type_display` is meant for human readable output, and `role_name` is not, no?

Reply:
Both are quite readable:
py:function
std:confval
py:class

Reply:
I think `configuration value` over `confval` is much better for non-technical users. We should definitely try to use the display value, that's the whole reason it exists if for this use.

Reply:
@ericholscher 
I realised (while testing) that some sphinx domains don't have values for type_display.
Should we still show for those who have? or just show role_name?